sql/insert.go

```diff
@@ -93,24 +115,27 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 		return nil, err
 	}
 	if err := p.db.Run(&b); err != nil {
+		if tErr, ok := err.(*proto.ConditionFailedError); ok {
+			return nil, fmt.Errorf("duplicate key value %q violates unique constraint %s", tErr.ActualValue.Bytes, "TODO(tamird)")
+		}
```

Comment:
Looks like there's a bug(?) in batch that prevents looping over `b.Results`. See #1891.